Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +212,49 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
+        $document = $this->dm->find(null, $path);
 
-        return $this->dm->find(null, $path);
+        if ($document instanceof AutoRouteInterface) {
+            return $document;
+        }
+
+        return null;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * Convert the given generic document to an auto route document, migrating any children which it may have.
+     *
+     * @param Generic $document
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
```

Comment:
@dbu I looked at ``DocumentMigrateClassCommand`` and its based on ``NodesUpdateCommand`` .. as such it doesn't make any sense to refactor the command.

Furthermore moving this to a helper class also does not seem to bring all that much benefit since its already all in a dedicated class .. I guess the only benefit would be abstracting the PHPCR session away from the PHPCR ODM user. Since it would still require fetching another class etc not sure if its worth it .. at the same time adding a new method on the DocumentManager for this also seems a bit overboard.

Furthermore, I guess technically what should also work instead of the ``$this->dm->getPhpcrSession()->save();`` call one could also do ``$this->dm->flush($document);`` but even if that works it would have more overhead and could trigger listeners which is likely not intended.

So to conclude, I think the current approach is the best way it can be already. Sorry for the noise :)